Expensify/react-native-x-maps
# Add workflow for auto publishing the repo to NPM

This PR implements a GitHub workflow that automatically publishes a new version of the library to NPM every time a PR is merged to `main`. This flow was copied from react-native-onyx.

# Tests
None. This can only be tested once it is merged. Then we will see if it works or not!

# QA
- Check the GH actions to see if it ran without any errors
- Look in https://www.npmjs.com/ to see if the package was properly published
